packages/strapi-design-system/src/SubNav/SubNavHeader.tsx

```diff
@@ -14,17 +13,39 @@ import { IconButton } from '../IconButton';
 import { Searchbar, SearchForm } from '../Searchbar';
 import { Typography } from '../Typography';
 
+export interface SubNavHeaderProps {
+  as: keyof HTMLElementTagNameMap;
```

Comment:
This is interesting, I didn't know about this type – where did you learn it? Should it be applied elsewhere in the codebase? :)

Reply:
I think it's incorrect looking at the typings for `styled-components`:

```ts
type StyledComponentPropsWithAs<
    C extends string | React.ComponentType<any>,
    T extends object,
    O extends object,
    A extends keyof any,
    AsC extends string | React.ComponentType<any> = C,
    FAsC extends string | React.ComponentType<any> = C
> = StyledComponentProps<C, T, O, A, FAsC> & { as?: AsC | undefined; forwardedAs?: FAsC | undefined };
```

Maybe all `as` calls should be `StyledComponentPropsWithAs['as']`? Also, it definitely is not required imo

Reply:
I've updated it to `string | React.ComponentType<any>`. I still think `keyof HTMLElementTagNameMap | React.ComponentType<any>` would be stricter, but keeping it inline with styled-components sounds good to me.